tiledb/sm/query/readers/aggregators/aggregate_buffer.h

```diff
@@ -0,0 +1,216 @@
+/**
+ * @file   aggregate_buffer.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class AggregateBuffer.
+ */
+
+#ifndef TILEDB_AGGREGATE_BUFFER_H
+#define TILEDB_AGGREGATE_BUFFER_H
+
+#include "tiledb/common/status.h"
+#include "tiledb/sm/enums/layout.h"
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+class ResultTile;
+
+class AggregateBuffer {
```

Comment:
Can't say I'm super excited to see yet another buffer class being added here. And this one now adds std::optional buffers where I think most of the others just rely one nullptr for things?

Reply:
This standardizes the way we pass in data to be aggregated... It contains all information for a field... Offsets, fixed data, var data, validity data, bitmap. Potentially something else than buffer could be used.